django/contrib/gis/db/models/fields.py

```diff
@@ -371,3 +392,37 @@ class ExtentField(GeoSelectFormatMixin, Field):
 
     def get_internal_type(self):
         return "ExtentField"
+
+
+class RasterField(BaseSpatialField):
+    """
+    Raster field for GeoDjango
+    """
+
+    description = "Raster Field"
+    geom_type = 'RASTER'
+
+    def _check_connection(self, connection):
```

Comment:
I added checks for the connection feature to `db_type` (which is called on table creation I think) and the `get_db_prep_value` (which is called on adding a field through migrations). There are tests for this exception in the [migrations tests](https://github.com/geodesign/django/blob/c9a5ce90aba708ea65b1263bb95fccb3e88cda75/tests/gis_tests/gis_migrations/test_operations.py#L126) and the [field tests](https://github.com/geodesign/django/blob/c9a5ce90aba708ea65b1263bb95fccb3e88cda75/tests/gis_tests/rasterapp/test_rasterfield.py#L85). With this, all the tests pass and the creation of raster fields on non-raster databases will raise an exception.